core/storage/base_model/gae_models.py

```diff
@@ -1501,7 +1600,10 @@ def get_snapshots_metadata(
             for snapshot_id in snapshot_ids]
         returned_models = datastore_services.get_multi(metadata_keys)
 
-        for ind, model in enumerate(returned_models):
+        # Here, the type of returned_models is List[Optional[Model]] but the
+        # type of model is strict VersionedModel, Thus to avoid MyPy assignment
+        # error, we added an ignore here.
+        for ind, model in enumerate(returned_models):  # type: ignore[assignment]
```

Comment:
Why does this need to be added?

Reply:
After adding overloads, the model's type is changed from `optional[VersionedModel]` to `VersionedModel`. But the type of returned_models is `List[Optional[Model]]`. So MyPy throws an error on considering the case where returned_models is `List[None]` and the type of model is `VersionedModel`.

Reply:
Ahhh, I see now. Let's fix this by naming the variable differently so that we do not reuse the `model` variable.